mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -199,8 +238,8 @@ class LinearRegression(override val uid: String)
 
     val costFun = new LeastSquaresCostFun(instances, yStd, yMean, $(fitIntercept),
       $(standardization), featuresStd, featuresMean, effectiveL2RegParam)
-
-    val optimizer = if ($(elasticNetParam) == 0.0 || effectiveRegParam == 0.0) {
```

Comment:
Owlqn is an generalized version of lbfgs but can solve L1. So when the solver is lbfgs, and L1 is used, we should use owlqn.